vmtkScripts/vmtkimagesmoothing.py

```diff
@@ -28,26 +29,32 @@ def __init__(self):
         
         self.Image = None
 
+        self.Method = 'gauss'
         self.StandardDeviation = 1.0
         self.RadiusFactor = 5.0
         self.Dimensionality = 3
+        self.Conductance = 1.0
+        self.NumberOfIterations = 5
+        self.TimeStep = 0.0625
```

Comment:
Not sure if this timestep is the best default. For the standard image we use in our test cases, I recieved the following output while using the default parameters. 

```
...
Automatic piping vmtkimagesmoothing
    Image = vmtkimagereader-0.Image
Parsing options vmtkimagesmoothing
    Method = anisotropic
Explicit piping vmtkimagesmoothing
Input vmtkimagesmoothing members:
    Id = 0
    Disabled = 0
    Image = vtkImageData
    ImageInputFileName =
    Method = anisotropic
    StandardDeviation = 1.0
    RadiusFactor = 5.0
    Dimensionality = 3
    Conductance = 1.0
    NumberOfIterations = 5
    TimeStep = 0.0625
    ImageOutputFileName =
Executing vmtkimagesmoothing ...
WARNING: In /Users/rick/anaconda3/envs/vmtkbuild/include/ITK-4.13/itkAnisotropicDiffusionImageFilter.hxx, line 82
GradientAnisotropicDiffusionImageFilter (0x7fc46cf20600): Anisotropic diffusion unstable time step: 0.0625
Stable time step for this image must be smaller than 0.0549316

WARNING: In /Users/rick/anaconda3/envs/vmtkbuild/include/ITK-4.13/itkAnisotropicDiffusionImageFilter.hxx, line 82
GradientAnisotropicDiffusionImageFilter (0x7fc46cf20600): Anisotropic diffusion unstable time step: 0.0625
Stable time step for this image must be smaller than 0.0549316

WARNING: In /Users/rick/anaconda3/envs/vmtkbuild/include/ITK-4.13/itkAnisotropicDiffusionImageFilter.hxx, line 82
GradientAnisotropicDiffusionImageFilter (0x7fc46cf20600): Anisotropic diffusion unstable time step: 0.0625
Stable time step for this image must be smaller than 0.0549316

WARNING: In /Users/rick/anaconda3/envs/vmtkbuild/include/ITK-4.13/itkAnisotropicDiffusionImageFilter.hxx, line 82
GradientAnisotropicDiffusionImageFilter (0x7fc46cf20600): Anisotropic diffusion unstable time step: 0.0625
Stable time step for this image must be smaller than 0.0549316

WARNING: In /Users/rick/anaconda3/envs/vmtkbuild/include/ITK-4.13/itkAnisotropicDiffusionImageFilter.hxx, line 82
GradientAnisotropicDiffusionImageFilter (0x7fc46cf20600): Anisotropic diffusion unstable time step: 0.0625
Stable time step for this image must be smaller than 0.0549316

Done executing vmtkimagesmoothing.
Output vmtkimagesmoothing members:
    Id = 0
    Image = vtkImageData
...
```

Reply:
The time step comes from (PixelSpacing)/2^{N+1}, so for 1.0/2^{4}=0.0625
This is the default in slicer, which I have used (obviously this warning message is hidden). I was thrown off at first, but there is really no way to know apriori what the user wants. We could have the script check for the largest dimension and calculate the minimum time step as the default and only override is the user specifies one?
the challenge here is that obfuscates the interface a little because the script will print out the default time step so another message would be required to inform the user that the timestep is now different.

Reply:
Ok. thanks for the clarification!

 Since I'm assuming that we want to use the correct value (calculated from the equation) for most people, in most cases, what if we included a method to `auto-calculate-timestep`, along with an option which defaulted to use that?  We could then return the used timestep as an output member to the script, ensuring that the user could backtrack and figure out which value was used (sort of like how isosurface value can be returned in http://www.vmtk.org/vmtkscripts/vmtkimageinitialization.html). 

If the `auto-calculate-timestep` option was enabled, it would just overwrite whatever value of `self.TimeStep` was provided in the `__init__` method. If it was disabled, we would use the `self.TimeStep` value provided by default, or override it automatically with a user defined value if they decided they wanted to tweak it manually. 

I think that would give us both customizability, and standardized logging in one go?

Reply:
fantastic idea, will implement this.